File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/Communication/NotificationSharedTo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.Communication
{
    public class NotificationSharedTo : AuditableEntity
    {
        [Key]
        public long ID { get; set; }
        public long UserID { get; set; }
        public bool NotifiedOrNot { get; set; }

        [ForeignKey("Notification")]
        public long NotificationID { get; set; }
        public Notification Notification { get; set; }

        [ForeignKey("UserType")]
        public long UserTypeID { get; set; }
        public UserType UserType { get; set; }
    }
}

File: LMS_CMS_Frontend/src/app/Models/Communication/notification-shared-to.ts
export class NotificationSharedTo {
     constructor(
        public id: number = 0,  
        public userId :number =0,
        public notifiedOrNot: boolean = false,
        public notificationID :number =0,
        public userTypeID :number =0,
        public imageLink: string = '', 
        public text: string = '',
        public link: string = '',
        public isAllowDismiss: boolean = false
    ) {}
}
 